training/src/anemoi/training/train/tasks/base.py

```diff
@@ -761,3 +765,22 @@ def configure_optimizers(self) -> tuple[list[torch.optim.Optimizer], list[dict]]
         )
 
         return [optimizer], [{"scheduler": scheduler, "interval": "step"}]
+
+    def setup(self, stage: str) -> None:
+        """Lightning hook that is called after model is initialized but before training starts."""
+        # The conditions should be separate, but are combined due to pre-commit hook
+        if stage == "fit" and self.trainer.is_global_zero and self.logger is not None:
+            # Log hyperparameters on rank 0
+            hyper_params = OmegaConf.to_container(convert_to_omegaconf(self.config), resolve=True)
+            hyper_params.update({"variable_loss_scaling": self._scaling_values_log})
+            # Expand keys for better visibility
+            expand_keys = OmegaConf.select(
```

Comment:
Didn't know you could do this with OmegaConf, just curious was this for readability or  did you encounter any issues if calling?
```
            self.logger.log_hyperparams(
                hyper_params,
                expand_keys=self.config.diagnostics.log.mlflow.expand_hyperparams,
            )
```